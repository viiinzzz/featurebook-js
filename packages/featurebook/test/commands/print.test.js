/* eslint-disable camelcase */
/* eslint-disable quotes */
const path = require('path');
const {
  Iwant, given, when, then_wait_at_most, expect, before, after,
  rootdir, node, cleanupFile,
  content_is_valid_pdf, length_at_least,
  one_minute,
} = require('../test.helpers');

const getOutputFile = (outputDir) => {
  const outputPath = outputDir.startsWith('.') ? path.join(rootdir, outputDir) : outputDir;
  return path.join(outputPath, 'specification.pdf');
};

given('I have a folder with .feature files', () => {
  const folder = '../features';
  const outputDir = `${folder}/dist/pdf`;
  const outputFile = getOutputFile(outputDir);
  when('I run command ‘print’', () => {
    then_wait_at_most(one_minute)
      .that('it generates a valid pdf file', async () => {
        const output = await node({
          program: 'bin/featurebook',
          options: `print -o ${outputDir} ${folder}`,
          outputFile,
        });

        Iwant(output)
          .and.to.satisfy(length_at_least(100000))
          .and.to.satisfy(content_is_valid_pdf);
      });

    const cleanup = async () => cleanupFile(outputFile);
    before(cleanup);
    after(cleanup);
  });
});

given('I have a folder with .feature files', () => {
  const folder = '../features.simple';
  const outputDir = `${folder}/dist/pdf`;
  const outputFile = getOutputFile(outputDir);
  when('I dry-run command ‘print’', () => {
    then_wait_at_most(one_minute).that('it generates expected output to the console', async () => {
      const str = await node({
        program: 'bin/featurebook',
        options: `print --dry-run -o ${outputDir} ${folder}`,
      });

      expect(str).toMatchSnapshot();
    });

    const cleanup = async () => cleanupFile(outputFile);
    before(cleanup);
    after(cleanup);
  });
});
