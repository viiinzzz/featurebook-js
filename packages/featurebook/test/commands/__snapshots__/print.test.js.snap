// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I have a folder with .feature files I dry-run command â€˜printâ€™ it generates expected output to the console 1`] = `
"featurebook v3.0.0-b.6 pdf:
---dry-run---
 docDefinition:
  defaultStyle:
    font: Anaheim
    fontSize: 12
  pageSize: A4
  info:
    Title: Untitled
  styles:
    header1:
      fontSize: 24
      bold: true
      marginBottom: 5
    header2:
      fontSize: 20
      bold: true
      marginBottom: 5
    header3:
      fontSize: 18
      bold: true
      marginBottom: 5
  content:
    - text: "undefined "
      style: header1
    - toc:
        title:
          text: Table of content
          style: header2
        numberStyle:
          bold: true
    - text: Features.simple
      style: header1
      margin:
        - 0
        - 20
        - 0
        - 10
      pageBreak: before
      tocItem: true
      tocStyle:
        fontSize: 14
      tocMargin:
        - 0
        - 10
        - 0
        - 0
    - table:
        headerRows: 0
        widths:
          - "*"
        body:
          - - - - text: Foreword
                  fontSize: 24
                  bold: true
                  marginBottom: 5
                  style:
                    - html-h1
              - - text: To be or not to be ...
                  style:
                    - html-p
      layout:
        defaultBorder: false
        fillColor: "#ddd"
    - text: Act 1
      style: header1
      margin:
        - 0
        - 20
        - 0
        - 10
      pageBreak: before
      tocItem: true
      tocStyle:
        fontSize: 14
      tocMargin:
        - 0
        - 10
        - 0
        - 0
    - table:
        headerRows: 0
        widths:
          - "*"
        body:
          - - - - text: Act 1 Relationships
                  fontSize: 24
                  bold: true
                  marginBottom: 5
                  style:
                    - html-h1
      layout:
        defaultBorder: false
        fillColor: "#ddd"
    - table:
        headerRows: 0
        widths:
          - "*"
        body:
          - - maxWidth: 505
              image: B-npmjs_published-featurebook-js-packages-features_simple-hamlet_jpg
      layout:
        defaultBorder: false
        fillColor: "#ddd"
    - table:
        headerRows: 0
        widths:
          - "*"
        body:
          - - maxWidth: 505
              image: cucumber
      layout:
        defaultBorder: false
        fillColor: "#ddd"
    - text:
        - text: "Feature:"
          color: red
          fontSize: 16
        - text: " Act 1 Scene 1"
          fontSize: 16
      margin:
        - 0
        - 10
        - 0
        - 5
    - text:
        - text: "Scenario:"
          color: red
          fontSize: 14
        - text: " â€˜tis but our fantasy"
          fontSize: 14
      margin:
        - 0
        - 10
        - 0
        - 5
    - text:
        - text: "Given:"
          color: red
        - text: " Marcellus and Bernardo"
    - text:
        - text: "When:"
          color: red
        - text: " Horatio comes"
    - text:
        - text: "Then:"
          color: red
        - text: " Marcellus informs"
  footer:
    - table:
        widths:
          - 500
        headerRows: 1
        body:
          - - text: ""
              border:
                - false
                - true
                - false
                - false
      layout: {}
      margin:
        - 40
        - 10
        - 0
        - 0
    - text: 10/31/2022
      margin:
        - 40
        - 0
        - 40
        - 0
      style: smallText
    - text: Untitled
      margin:
        - 0
        - 5
        - 40
        - 0
      alignment: right
      style: smallText
      bold: true
  images:
    cucumber: data:image/png;base64,...
    B-npmjs_published-featurebook-js-packages-features_simple-hamlet_jpg:
      data:image;base64,...

[90m..\\features.simple\\dist\\pdf\\specification.pdf[39m
done.
"
`;
